sensor_test.py

```diff
@@ -0,0 +1,167 @@
+
+
+def test_set_sensor_name(get_sensor_info, set_sensor_name):
+    """
+    1. Set sensor name to "new_name".
+    2. Get sensor_info.
+    3. Validate that current sensor name matches the name set in Step 1.
+    """
+    print("Step 1", 'Set sensor name to "new_name"')
+    updated_sensor_name = "new_name"
+    set_sensor_name(updated_sensor_name)
+
+    print("Step 2", 'Get sensor_info')
+    sensor_info_after_name_change = get_sensor_info()
+
+    print("Step 3", 'Validate that current sensor name matches the name set in Step 1')
+    assert sensor_info_after_name_change.get("name") == updated_sensor_name
+
+
+def test_set_sensor_reading_interval(
+        get_sensor_info, set_sensor_reading_interval, get_sensor_reading
+):
+    """
+    1. Set sensor reading interval to 1.
+    2. Get sensor info.
+    3. Validate that sensor reading interval is set to interval from Step 1.
+    4. Get sensor reading.
+    5. Wait for interval specified in Step 1.
+    6. Get sensor reading.
+    7. Validate that reading from Step 4 doesn't equal reading from Step 6.
+    """
+    print("Step 1", "Set sensor reading interval to 1")
+    new_sensor_reading_interval = 1
+    set_sensor_reading_interval(new_sensor_reading_interval)
+
+    print("Step 2", "Get sensor info")
+    sensor_info_after_reading_interval_change = get_sensor_info()
+
+    print("Step 3", "Validate that sensor reading interval is set to interval from Step 1")
+    assert sensor_info_after_reading_interval_change.get("reading_interval") == new_sensor_reading_interval
+
+    print("Step 4", "Get sensor reading")
+    sensor_reading_before_wait = get_sensor_reading()
+
+    print("Step 5", "Wait for interval specified in Step 1")
+    time.sleep(new_sensor_reading_interval)
+
+    print("Step 6", "Get sensor reading")
+    sensor_reading_after_wait = get_sensor_reading()
+
+    print("Step 7", "Validate that reading from Step 4 doesn't equal reading from Step 6")
+    assert sensor_reading_after_wait != sensor_reading_before_wait
+
+
+# Максимальна версія прошивки сенсора -- 15
+def test_update_sensor_firmware(get_sensor_info, update_sensor_firmware):
+    """
+    1. Get original sensor firmware version.
+    2. Request firmware update.
+    3. Get current sensor firmware version.
+    4. Validate that current firmware version is +1 to original firmware version.
+    5. Repeat steps 1-4 until sensor is at max_firmware_version - 1.
+    6. Update sensor to max firmware version.
+    7. Validate that sensor is at max firmware version.
+    8. Request another firmware update.
+    9. Validate that sensor doesn't update and responds appropriately.
+    10. Validate that sensor firmware version doesn't change if it's at maximum value.
+    """
+    max_firmware_version = 15
+
+    print("Step 1" "Get original sensor firmware version")
+    current_sensor_fw_ver = get_sensor_info().get("firmware_version")
+
+    print("Step 2" "Request firmware update"
```

Comment:
2 things about this statement:
1. It outputs just the first line to the console (at least on my computer).
2. These would be better off split and put near the actions in the code they correspond to.

For example:
```
while current_sensor_fw_ver < max_firmware_version - 1:
        sensor_fw_ver_before_update = current_sensor_fw_ver  # Variable was added to satisfy the requirement in Step 4

        print("Request firmware update")
        sensor_fw_upgrade_request = update_sensor_firmware()
        assert sensor_fw_upgrade_request == "updating"

        print("Get current firmware version")
        current_sensor_fw_ver = wait(
            func=lambda: get_sensor_info().get("firmware_version"),
            condition=lambda x: isinstance(x, int), tries=15, timeout=3)

        print("Validate that current firmware version is +1 to original firmware version")
        assert current_sensor_fw_ver == (sensor_fw_ver_before_update + 1)
```
It's not mandatory to follow test steps literally if it makes sense to do otherwise in the code. Description of step 5 is not really useful to put in the test because we'll be getting repeated prints in the loop. It also helps to separate steps with whitespace, so it's easier to read them.

Reply:
I haven't do that, cause i thought it be wrong to paste print in cycle. But you right it even more easy to read in this way.
and as I understand should I totaly remove step5 print from the test ? cause if put in before start of the cycle it confuses a litle bit